plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/PostgreDataSourceProvider.java

```diff
@@ -72,6 +72,8 @@ public long getFeatures() {
     @Override
     public String getConnectionURL(DBPDriver driver, DBPConnectionConfiguration connectionInfo) {
         DBAAuthModel<?> authModel = connectionInfo.getAuthModel();
+        String databaseName = connectionInfo.getDatabaseName();
+        connectionInfo.setDatabaseName(databaseName.replace("/", "%2F"));
```

Comment:
You cannot modify connection info in getter. 
It may lead to side effects. And you don't need to change it actually.
Just use fixed name for URL build